### PR TITLE
doxygen + wire + travis + readme

### PR DESCRIPTION
**Pull request by hoffmannjan**

### CODE REVIEW

**Comment by hoffmannjan** [comment]
@ladyada don't have access to add `gh-pages`

**Comment by ladyada** [comment]
access added!

File: Adafruit_CAP1188.cpp [hunk 1]
```diff
@@ -63,24 +103,34 @@ Adafruit_CAP1188::Adafruit_CAP1188(int8_t clkpin, int8_t misopin,
...
     _i2caddr = i2caddr;
+    _wire->begin();
   } else if (_clk == -1) {
     // Hardware SPI
     pinMode(_cs, OUTPUT);
     digitalWrite(_cs, HIGH);
 #ifdef SPI_HAS_TRANSACTION
-      SPI.begin();
+    SPI.begin();
 #else
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: honestly we shouldn't support SPI with transaction anymore, please remove all sections that are not SPI_HAS_TRANSACTION

File: Adafruit_CAP1188.cpp [hunk 2]
```diff
@@ -102,105 +152,113 @@ boolean Adafruit_CAP1188::begin(uint8_t i2caddr) {
...
   if (t) {
     writeRegister(CAP1188_MAIN, readRegister(CAP1188_MAIN) & ~CAP1188_MAIN_INT);
   }
   return t;
 }
 
+/*!
+ *   @brief  Controls the output polarity of LEDs.
+ *   @param  x
+ *           Writes x to CAP1188_LEDPOL register
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: what does x mean (true/false)? maybe change x to a more useful name

File: Adafruit_CAP1188.cpp [hunk 3]
```diff
@@ -102,105 +152,113 @@ boolean Adafruit_CAP1188::begin(uint8_t i2caddr) {
...
   writeRegister(CAP1188_STANDBYCFG, 0x30);
   return true;
 }
 
-uint8_t  Adafruit_CAP1188::touched(void) {
+/*!
+ *   @brief  Reads from CAP1188_SENINPUTSTATUS and if touch is detected writes to CAP1188_MAIN
+ *   @return Returns read from CAP1188_SENINPUTSTATUS
+ */
+uint8_t Adafruit_CAP1188::touched() {
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: this function reads the touched status, returns an 8-bit value where the 0 pad is the LSB (1 if touched, 0 if not)

File: Adafruit_CAP1188.cpp [hunk 4]
```diff
@@ -102,105 +152,113 @@ boolean Adafruit_CAP1188::begin(uint8_t i2caddr) {
...
-static uint8_t i2cread(void) {
-  #if ARDUINO >= 100
-  return Wire.read();
-  #else
-  return Wire.receive();
-  #endif
+ *   @brief  Abstract away platform differences in Arduino wire library
+ *   @return Read value 
+ */
+uint8_t Adafruit_CAP1188::i2cread() {
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: remove this, we don't support ARDUINO before 100, we should just use `read()`

File: Adafruit_CAP1188.cpp [hunk 5]
```diff
@@ -1,48 +1,74 @@
...
+ *            number of pin where reset is connected
+ *
+ */
 Adafruit_CAP1188::Adafruit_CAP1188(int8_t resetpin) {
   // I2C
   _resetpin = resetpin;
   _i2c = true;
 }
 
+/*!
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: add optional SPI pointer device (default to &SPI) so they can pass in optional SPI device

Comment (hoffmannjan) [reply Adafruit_CAP1188.cpp]: do we have any library that have optional spi pointer param? would be cool if we clean up this to be consistent.

**Comment by hoffmannjan** [comment]
@ladyada waiting for your reply about SPI and I think all is fixes as you requested :)

File: Adafruit_CAP1188.cpp (hunk 5 above)
Comment (hoffmannjan) [reply Adafruit_CAP1188.cpp]: OK, please look is it done right :).
Also, imho we can remove all `SPI_HAS_TRANSACTION` from the code and add one which just throws error if `!SPI_HAS_TRANSACTION` but looking forward for your approve :)

File: Adafruit_CAP1188.h [hunk 6]
```diff
@@ -1,63 +1,96 @@
...
+
+private:
+  void spi_begin();
+  void spi_end();
 
- private:
   uint8_t spixfer(uint8_t x);
   boolean _i2c;
   int8_t _i2caddr, _resetpin, _cs, _clk, _mosi, _miso;
+  uint8_t mySPCR, SPCRback;
```
Comment (ladyada) [inline Adafruit_CAP1188.h]: remove, this isnt used anymore

File: Adafruit_CAP1188.h [hunk 7]
```diff
@@ -1,63 +1,96 @@
...
+ *
+ *  BSD license, all text above must be included in any redistribution
+ */
 
 #if (ARDUINO >= 100)
- #include "Arduino.h"
+#include "Arduino.h"
 #else
- #include "WProgram.h"
+#include "WProgram.h"
```
Comment (ladyada) [inline Adafruit_CAP1188.h]: remove the if/else so we only support Arduino >= 100 now

File: Adafruit_CAP1188.cpp [hunk 8]
```diff
@@ -258,11 +282,7 @@ void Adafruit_CAP1188::writeRegister(uint8_t reg, uint8_t value) {
     spixfer(value);
     digitalWrite(_cs, HIGH);
 #ifdef SPI_HAS_TRANSACTION
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: remove the ifdef, we ONLY support if SPI has transactions

File: Adafruit_CAP1188.cpp [hunk 9]
```diff
@@ -214,39 +246,31 @@ uint8_t Adafruit_CAP1188::readRegister(uint8_t reg) {
...
 void Adafruit_CAP1188::writeRegister(uint8_t reg, uint8_t value) {
   if (_i2c) {
-    Wire.beginTransmission(_i2caddr);
+    _wire->beginTransmission(_i2caddr);
     i2cwrite((uint8_t)reg);
     i2cwrite((uint8_t)(value));
-    Wire.endTransmission();
+    _wire->endTransmission();
   } else {
 #ifdef SPI_HAS_TRANSACTION
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: remove

File: Adafruit_CAP1188.cpp [hunk 10]
```diff
@@ -214,39 +246,31 @@ uint8_t Adafruit_CAP1188::readRegister(uint8_t reg) {
     digitalWrite(_cs, HIGH);
     digitalWrite(_cs, LOW);
     spixfer(0x7F);
-    uint8_t reply = spixfer(0); 
+    uint8_t reply = spixfer(0);
     digitalWrite(_cs, HIGH);
 #ifdef SPI_HAS_TRANSACTION
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: remove

File: Adafruit_CAP1188.cpp [hunk 11]
```diff
@@ -102,110 +141,103 @@ boolean Adafruit_CAP1188::begin(uint8_t i2caddr) {
...
+    _wire->beginTransmission(_i2caddr);
     i2cwrite(reg);
-    Wire.endTransmission();
-    Wire.requestFrom(_i2caddr, 1);
-    return (i2cread());
+    _wire->endTransmission();
+    _wire->requestFrom(_i2caddr, 1);
+    return (_wire->read());
   } else {
 #ifdef SPI_HAS_TRANSACTION
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: remove

File: Adafruit_CAP1188.cpp [hunk 12]
```diff
@@ -102,110 +141,103 @@ boolean Adafruit_CAP1188::begin(uint8_t i2caddr) {
...
-  #else
-  Wire.send(x);
-  #endif
+    @param  x
+ */
+void Adafruit_CAP1188::i2cwrite(uint8_t x) {
+#if ARDUINO >= 100
+  _wire->write((uint8_t)x);
+#else
+  _wire->send(x);
```
Comment (ladyada) [inline Adafruit_CAP1188.cpp]: remove

**Comment by ladyada** [comment]
remove any old arduino (< 100) ifdefs and also only support SPI with transactions so remove the #if checks

File: Adafruit_CAP1188.h (hunk 7 above)
Comment (hoffmannjan) [reply Adafruit_CAP1188.h]: in all libs?

Comment (ladyada) [reply Adafruit_CAP1188.h]: yes its ancient, WProgram is no longer used

**Pull request merged**